src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -63,6 +86,18 @@ public virtual Task<BaseConfiguration> GetBaseConfigurationAsync(CancellationTok
             throw new NotImplementedException();
         }
 
+        /// <summary>
+        /// Gets all valid last known good configurations.
+        /// </summary>
+        /// <returns>A collection of all valid last known good configurations.</returns>
+        internal ICollection<BaseConfiguration> GetValidLkgConfiguraitons()
+        {
+            if (_lkgConfigurationCache == null)
+                return null;
+
+            return _lkgConfigurationCache.ToArray().Where(x => x.Value.Value > DateTime.UtcNow).Select(x => x.Key).ToArray();
```

Comment:
The EventBasedLRUCache will keep the cache the right size and remove expired items in the background.
Could we just return _LkgConfigurationCache.ToArray() assuming all items in the cache should be tried?